linera-chain/src/certificate.rs

```diff
@@ -69,13 +75,25 @@ impl<T: Debug> Debug for GenericCertificate<T> {
     }
 }
 
-// NOTE: For backwards compatiblity reasons we serialize the new `Certificate` type as the old
-// one. Otherwise we would be breaking the RPC API schemas. We can't implement generic serialization
-// for `Certificate<T>` since only specific `T`s have corresponding `CertificateValue` variants.
 impl Serialize for ValidatedBlockCertificate {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let cert = Certificate::from(self.clone());
-        cert.serialize(serializer)
+        let mut state = serializer.serialize_struct("ValidatedBlockCertificate", 4)?;
+        state.serialize_field("hash", &self.value.hash())?;
+        state.serialize_field("value", &self.inner())?;
+        state.serialize_field("round", &self.round)?;
+        state.serialize_field("signatures", &self.signatures)?;
+        state.end()
+    }
+}
+
+impl Serialize for TimeoutCertificate {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut state = serializer.serialize_struct("TimeoutCertificate", 4)?;
+        state.serialize_field("hash", &self.value.hash())?;
+        state.serialize_field("value", &self.inner())?;
+        state.serialize_field("round", &self.round)?;
+        state.serialize_field("signatures", &self.signatures)?;
+        state.end()
```

Comment:
Does that really BCS-serialize to the same bytes as a `Certificate` with `Timeout` value?

Reply:
Does it matter? I carry-on the hash and I map to/from the old `ConfirmedValue` types.

Reply:
I guess it will only matter in the future, when we use this directly to compute the hash.

Or at least it will matter then that `ConfirmedBlockCertificate` serializes to something different than `ValidatedBlockCertificate`.